web/src/beta/components/SubTabButtonList/index.tsx

```diff
@@ -0,0 +1,53 @@
+import type { FC } from "react";
+
+import { styled, useTheme, PublishTheme } from "@reearth/services/theme";
+
+import useManageSwitchState, { SwitchField } from "../../hooks/useManageSwitchState/hooks";
+import Text from "../Text";
+
+type CustomField = {
+  name: string;
+};
+type SwitchCustomField = SwitchField<CustomField>;
+
+export type Props = {
+  publishedTheme?: PublishTheme;
+  items: SwitchCustomField[];
+  onChange?: (id: string) => void;
+};
+const SubTabButtonList: FC<Props> = ({ publishedTheme, items, onChange }) => {
+  const theme = useTheme();
+  const { handleActivate, fields } = useManageSwitchState({ fields: items });
```

Comment:
I guess this hooks should run on the parent component.
then it will be used like `<SubButtonList items={fields} onChange={handleActive} />` and can delete `handleClick` in this component.

Reply:
Fix completed.